lib/charms/mongodb/v1/mongodb_provider.py

```diff
@@ -260,11 +260,27 @@ def update_app_relation_data(self) -> None:
                     config.uri,
                 )
 
+    def _get_or_set_password(self, relation: Relation) -> str:
+        """Retrieve password from cache or generate a new one.
+
+        Args:
+            relation (Relation): The relation for each the password is cached.
+
+        Returns:
+            str: The password.
+        """
+        password = self.database_provides.fetch_my_relation_field(relation.id, "password")
+        if password:
+            return password
+        password = generate_password()
+        self.database_provides.update_relation_data(relation.id, {"password": password})
```

Comment:
noob question, we use external juju secrets, why update `relation_data`?

Reply:
we need to setup password to secrets, and because of new version of the library I need to update it